File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        triangle=[]
        n=rowIndex+1
        for i in range(n):
            row=[1]*(i+1)
            for j in range(1,i):
                row[j]=triangle[i-1][j]+triangle[i-1][j-1]
            triangle.append(row)
        return triangle[n-1] 

File: 119-pascals-triangle-ii/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> triangle;
        int n=rowIndex+1;
        for(int i=0;i<n;i++){
            vector<int> row(i+1,1);
            for(int j=1;j<i;j++){
                row[j]=triangle[i-1][j]+triangle[i-1][j-1]; //prevrows elmnt [i-1][j] + the one on its left [i-1][j-1]
            }
            triangle.push_back(row);
        }
        return triangle[rowIndex];
    }
};